src/webgl/light.js

```diff
@@ -39,32 +39,16 @@ var p5 = require('../core/core');
  *
  */
 p5.prototype.ambientLight = function(v1, v2, v3, a){
-  var gl = this._renderer.GL;
-  var shaderProgram = this._renderer._getShader(
-    'lightVert', 'lightTextureFrag');
-
-  gl.useProgram(shaderProgram);
-  shaderProgram.uAmbientColor = gl.getUniformLocation(
-    shaderProgram,
-    'uAmbientColor[' + this._renderer.ambientLightCount + ']');
-
   var color = this._renderer._pInst.color.apply(
     this._renderer._pInst, arguments);
-  var colors = color._array;
-
-  gl.uniform3f( shaderProgram.uAmbientColor,
-    colors[0], colors[1], colors[2]);
+  var colors = color._array.slice(0, 3);
 
-  //in case there's no material color for the geometry
-  shaderProgram.uMaterialColor = gl.getUniformLocation(
-    shaderProgram, 'uMaterialColor' );
-  gl.uniform4f( shaderProgram.uMaterialColor, 1, 1, 1, 1);
+  this._renderer._setUniform('uAmbientColor[' +
```

Comment:
Hmm, I'm not sure if I see the problems of explicitly checking whether the shader program exists and if not, create one, then apply the uniform.

I think you're right that having a clear distinction between a global uniform setting vs. program specific uniform set is very important, but also setting global uniforms in a p5.Shader instance could be dangerous and potentially confusing since we really want to expose as much as the p5.Shader api as possible to the user.